src/unsafe_string.rs

```diff
@@ -54,6 +53,16 @@ trait HeaderMut<'a>: ThinMutExt<'a, Header> {
 impl<'a, T: ThinRefExt<'a, Header>> HeaderRef<'a> for T {}
 impl<'a, T: ThinMutExt<'a, Header>> HeaderMut<'a> for T {}
 
+// Constants for inline string storage
+const INLINE_STRING_MAX_LEN: usize = 7; // 7 bytes + null terminator = 8 bytes total
+
+/// Check if a string can be stored inline (null-terminated in pointer)
+fn can_inline_string(s: &str) -> bool {
+    let bytes = s.as_bytes();
+    // We need space for string bytes + null terminator in the upper bits (61 bits = 7.625 bytes)
+    bytes.len() <= INLINE_STRING_MAX_LEN && s.is_char_boundary(bytes.len())
```

Comment:
is the second condition not true by construction? how would `!s.is_char_boundary(s.as_bytes().len())`?